server/routes/api/v1/petTypesRouter.js

```diff
@@ -8,18 +10,57 @@ petTypesRouter.get('/', async (req, res) => {
     const petTypes = await PetType.findAll()
     res.status(200).json({ petTypes: petTypes })
   } catch (error) {
-    return res.status(500).json({ errors: error })
+    res.status(500).json({ errors: error })
   }
 })
 
 petTypesRouter.get('/:petType', async (req, res) => {
   try {
     const petType = await PetType.findByType(req.params.petType)
-    petType.adoptablePets = await petType.adoptablePets()
+    petType.adoptablePets = await petType.getAvailablePets()
     res.status(200).json({ pets: petType })
   } catch (error) {
-    return res.status(500).json({ errors: error })
+    res.status(500).json({ errors: error })
   }
 })
 
+petTypesRouter.get('/:petType/:adoptablePetId', async (req, res) => {
+  try {
+    const petType = await PetType.findByType(req.params.petType)
+    const adoptablePetId = req.params.adoptablePetId
+    const pet = await petType.findPet(adoptablePetId)
+    res.status(200).json({ pet: pet })
+  } catch (error) {
+    res.status(404).json({ errors: error })
+  }
+})
+
+petTypesRouter.post('/:petType', async (req, res) => {
+  try {
+    const petType = await PetType.findByType(req.body.petType)
+    
+    const adoptablePet = new AdoptablePet({ 
```

Comment:
this endpoint should instead be within something like an adoptable pet router since we're not primarily interacting with PetTypes